public/contextMenu.js

```diff
@@ -0,0 +1,258 @@
+      notDeletedPreviousElement = toDelete ? notDeletedPreviousElement : menuItem;
+      return !toDelete;
+    });
+};
+
+const create = (win, options) => {
+  const handleContextMenu = (event, props) => {
+    if (typeof options.shouldShowMenu === 'function' && options.shouldShowMenu(event, props) === false) {
+      return;
+    }
+
+    const {editFlags} = props;
+    const hasText = props.selectionText.trim().length > 0;
+    const can = type => editFlags[`can${type}`] && hasText;
+
+    const defaultActions = {
+      separator: () => ({type: 'separator'}),
+      cut: decorateMenuItem({
+        id: 'cut',
+        label: 'Cu&t',
+        enabled: can('Cut'),
+        visible: props.isEditable,
+        click(menuItem) {
+          const target = webContents(win);
+
+          if (!menuItem.transform && target) {
+            target.cut();
+          } else {
+            props.selectionText = menuItem.transform ? menuItem.transform(props.selectionText) : props.selectionText;
+            electron.clipboard.writeText(props.selectionText);
+          }
+        }
+      }),
+      copy: decorateMenuItem({
+        id: 'copy',
+        label: '&Copy',
+        enabled: can('Copy'),
+        visible: props.isEditable || hasText,
+        click(menuItem) {
+          const target = webContents(win);
+
+          if (!menuItem.transform && target) {
+            target.copy();
+          } else {
+            props.selectionText = menuItem.transform ? menuItem.transform(props.selectionText) : props.selectionText;
+            electron.clipboard.writeText(props.selectionText);
+          }
+        }
+      }),
+      paste: decorateMenuItem({
+        id: 'paste',
+        label: '&Paste',
+        enabled: editFlags.canPaste,
+        visible: props.isEditable,
+        click(menuItem) {
+          const target = webContents(win);
+
+          if (menuItem.transform) {
+            let clipboardContent = electron.clipboard.readText(props.selectionText);
+            clipboardContent = menuItem.transform ? menuItem.transform(clipboardContent) : clipboardContent;
+            target.insertText(clipboardContent);
+          } else {
+            target.paste();
+          }
+        }
+      }),
+      inspect: () => ({
+        id: 'inspect',
+        label: 'I&nspect Element',
+        click() {
+          win.inspectElement(props.x, props.y);
+
+          if (webContents(win).isDevToolsOpened()) {
+            webContents(win).devToolsWebContents.focus();
+          }
+        }
+      }),
+    };
+
+    const shouldShowInspectElement = typeof options.showInspectElement === 'boolean' ? options.showInspectElement : false;
+
+    let dictionarySuggestions = [];
+
+    let menuTemplate = [
+      defaultActions.cut(),
+      defaultActions.copy(),
+      defaultActions.paste(),
+      defaultActions.separator(),
+      shouldShowInspectElement && defaultActions.inspect(),
+      defaultActions.separator()
```

Comment:
If don't should the inspect element, then show two separators in a row?

Reply:
The "removeUsedMenuItems" had a logic to avoid two consecutive separators or a separator as a first/last element.
Since we also removed dynamic menuItems, this was further simplified.